crates/sui-types/src/messages_consensus.rs

```diff
@@ -43,6 +44,27 @@ pub struct ConsensusCommitPrologueV2 {
     pub consensus_commit_digest: ConsensusCommitDigest,
 }
 
+/// Uses an enum to allow for future expansion of the ConsensusDeterminedVersionAssignments.
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize, JsonSchema)]
+pub enum ConsensusDeterminedVersionAssignments {
+    // Cancelled transaction version assignment.
+    CancelledTransactions(Vec<(TransactionDigest, Vec<(ObjectID, SequenceNumber)>)>),
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+pub struct ConsensusCommitPrologueV3 {
+    /// Epoch of the commit prologue transaction
+    pub epoch: u64,
+    /// Consensus round of the commit
+    pub round: u64,
```

Comment:
while we're here, I just want to confirm with @mwtian that there are no plans to have multiple commits per round? If there's any chance of that, lets add a `sub_dag_index: Option<u64>` while we are at it.

Reply:
We don't want to remove the possibility. Yes let's have a subdag / commit index here.